crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1700,6 +1710,18 @@ impl World {
     pub fn init_resource<R: Resource + FromWorld>(&mut self) -> ComponentId {
         let caller = MaybeLocation::caller();
         let component_id = self.components_registrator().register_resource::<R>();
+
+        if !self
+            .components
+            .resource_entities
+            .contains_key(&component_id)
+        {
+            let entity = self.spawn(ResourceEntity::<R>::default()).id();
+            self.components
+                .resource_entities
+                .insert(component_id, entity);
+        }
+
```

Comment:
I think these could use `entry` to avoid the double lookup.  
```suggestion
        self.components
            .resource_entities
            .entry(component_id)
            .or_insert_with(|| self.spawn(ResourceEntity::<R>::default()).id());
```

Reply:
I hate to be the one relaying information from the rust borrow checker, but you can't borrow `self` for `self.spawn` if it's already borrowed for `self.components.etc`.

Reply:
> I hate to be the one relaying information from the rust borrow checker, but you can't borrow `self` for `self.spawn` if it's already borrowed for `self.components.etc`.

Oh, hah, sorry!  

(Wow, I'm used to those being false positives because the method doesn't actually touch the collection, but in this case it's right because an observer could modify `components`!)